backend/src/main/scala/cromwell/backend/callcaching/CacheHitDuplicating.scala

```diff
@@ -0,0 +1,100 @@
+package cromwell.backend.callcaching
+
+import java.nio.file.Path
+
+import akka.actor.ActorRef
+import cromwell.backend.BackendCacheHitCopyingActor
+import cromwell.backend.BackendJobExecutionActor.{BackendJobExecutionResponse, SucceededResponse}
+import cromwell.backend.io.JobPaths
+import cromwell.core.PathCopier
+import cromwell.core.simpleton.{WdlValueBuilder, WdlValueSimpleton}
+import wdl4s.values.WdlFile
+
+/**
+  * Mixin implementing common functionality for a BackendCacheHitCopyingActor.
+  *
+  * Implements copyCachedOutputs, with abstract methods for converting a string to a path, duplicating a path, returning
+  * a reference to the service registry actor, and obtaining various metadata/outputs for the job.
+  */
+trait CacheHitDuplicating {
+  this: BackendCacheHitCopyingActor =>
+
+  /**
+    * Duplicates two paths returned by getPath.
+    *
+    * @param source      Source path.
+    * @param destination Destination path.
+    */
+  protected def duplicate(source: Path, destination: Path): Unit
+
+  /**
+    * Returns an absolute path to the file.
+    *
+    * NOTE: If necessary for separate credentialing, we may split this method into getSourcePath and getDestinationPath.
+    *
+    * @param file the string version of the path
+    * @return an absolute path to the file with potential credentials embedded within.
+    */
+  protected def getPath(file: String): Path
+
+  protected def destinationCallRootPath: Path
+
+  protected def serviceRegistryActor: ActorRef
+
+  protected def destinationJobDetritusPaths: Map[String, Path]
+
+  // Usually implemented by a subclass of JobCachingActorHelper
+  protected def metadataKeyValues: Map[String, Any]
+
+  private def lookupSourceCallRootPath(sourceJobDetritusFiles: Map[String, String]): Path = {
+    sourceJobDetritusFiles.get(JobPaths.CallRootPathKey).map(getPath).getOrElse(throw new RuntimeException(
+      s"The call detritus files for source cache hit aren't found for call ${jobDescriptor.call.fullyQualifiedName}")
```

Comment:
Could this exception be turned to a `FailedResponse` instead and sent back instead of the `SucceededResponse` in `copyCachedOutputs` ? Or at least are we sure that crashing the actor this way wouldn't hang the execution ?
(I know this is an edge "will never happen" case anyway but..)